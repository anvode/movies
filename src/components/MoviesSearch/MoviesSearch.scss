input {

    &.input__field {
        background-clip: padding-box;
        background-color: var(--color-white);
        border: 1px solid var(--color-base);
        border-radius: 0;
        color: var(--color-base);
        display: block;
        font-size: 1rem;
        line-height: 1.47;
        padding: 15px 20px;
        width: 100%;
    }

    &:focus {
        border-color: var(--color-primary);
        box-shadow: 0 0 0 0.2rem var(--color-primary);
        outline: 0;
    }

    &::-ms-expand {
        background-color: transparent;
        border: 0;
    }

    &::placeholder {
        color: var(--color-gray-light);
    }

    &:disabled,
    &[readonly] {
        background-color: var(--color-gray);
    }
}

.btn {
    background: var(--color-primary);
    border: 0;
    border-radius: 0 0 5px 5px;
    box-shadow: 0 6px var(--color-base);
    color: var(--color-base);
    cursor: pointer;
    display: inline-block;
    font-family: inherit;
    font-size: inherit;
    font-weight: 700;
    letter-spacing: 1px;
    margin: 0;
    outline: none;
    padding: 15px 80px;
    position: relative;
    text-transform: uppercase;
    transition: all 0.3s;

    &:hover,
    &:focus {
        background: var(--color-base);
        box-shadow: 0 6px var(--color-primary);
        color: var(--color-white);
    }
}

.search {
    &__button {
        flex: 1 1 auto;
        margin-bottom: 0.5rem;
        padding: 0 0.8rem;
    }

    &__input {
        flex: 1 1 auto;
        margin-bottom: 0.5rem;
        padding: 0 0.8rem;
    }
}
