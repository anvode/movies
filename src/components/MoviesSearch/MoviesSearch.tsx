import * as React from 'react';
import { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';

import { RootState } from '../../redux/root-reducer';
import { fetchMovies } from '../../redux/movies/movies.actions';

import './MoviesSearch.scss';

export interface MoviesSearchProps {}

const MoviesSearch: React.FC<MoviesSearchProps> = () => {
    const {searchQuery} = useSelector((state: RootState) => state.movies);
    const dispatch = useDispatch();
    const [state, setState] = useState(searchQuery);

    const changeHandler = (event: React.ChangeEvent<HTMLInputElement>): void => {
        setState(event.target.value);
    };

    const submitHandler = (event: React.FormEvent<HTMLFormElement> | React.MouseEvent<HTMLButtonElement, MouseEvent>): void => {
        event.preventDefault();
        !!state.length && dispatch(fetchMovies(state, 1));
    };

    return <>
        <h1 className="search__title">Search for: {searchQuery}</h1>
        <form onSubmit={submitHandler} className="search__form">
            <div className="row">
                <div className="search__input input">
                    <input
                        type="text"
                        aria-label="Suchbegriff eingeben (zB: Bloodsport)"
                        onChange={changeHandler}
                        placeholder="Suchbegriff eingeben (zB: Bloodsport)"
                        id="search"
                        name="search"
                        value={state}
                        className="search__input input__field">
                    </input>
                </div>
                <div className="search__button">
                    <button onClick={submitHandler} type="submit" className="btn">Search</button>
                </div>
            </div>
        </form>
    </>;
};

export default MoviesSearch;