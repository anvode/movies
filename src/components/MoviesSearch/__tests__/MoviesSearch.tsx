import React from 'react';
import Store from '../../../redux/store';
import { render } from '@testing-library/react';
import MoviesSearch from '../MoviesSearch';
import {Provider} from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

describe('MoviesSearch', () => {
    it('renders without crashing', () => {
        const { asFragment } = render( <Provider store={Store}><BrowserRouter><MoviesSearch /></BrowserRouter></Provider>);
        expect(asFragment()).toMatchSnapshot();
    });
});