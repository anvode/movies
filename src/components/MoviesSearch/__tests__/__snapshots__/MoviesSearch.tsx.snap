// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesSearch renders without crashing 1`] = `
<DocumentFragment>
  <h1
    class="search__title"
  >
    Search for: 
  </h1>
  <form
    class="search__form"
  >
    <div
      class="row"
    >
      <div
        class="search__input input"
      >
        <input
          aria-label="Suchbegriff eingeben (zB: Bloodsport)"
          class="search__input input__field"
          id="search"
          name="search"
          placeholder="Suchbegriff eingeben (zB: Bloodsport)"
          type="text"
          value=""
        />
      </div>
      <div
        class="search__button"
      >
        <button
          class="btn"
          type="button"
        >
          Search
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
